### PR TITLE
Add FAQ for MultiprocessIterator + OpenCV problem

### LINKED ISSUE #4586: Add a workaround of the problem around OpenCV and MultiprocessIterator deadlock to "Tips and FAQs" page in the documentation
Workaround for the problem caused by OpenCV when using `MultiprocessIterator` (Related to #2903, https://github.com/opencv/opencv/issues/5150)

### PR DESCRIPTION
**Pull request by mitmul**
This PR resolves #4586

### CODE REVIEW

File: docs/source/tips.rst [hunk 1]
```diff
@@ -105,3 +105,17 @@ Now your model is accelerated by iDeep!
 
 Please note that not all functions and optimizers support iDeep acceleration.
 Also note that iDeep will not be used depending on the shape and data type of the input data.
+
+My training process gets stucked when using MultiprocessIterator
```
Comment (kmaehashi) [inline docs/source/tips.rst]: s/stucked/stuck/g
(ditto for the contents)

File: docs/source/tips.rst [hunk 2]
```diff
@@ -105,3 +105,17 @@ Now your model is accelerated by iDeep!
...
+----------------------------------------------------------------
+
+When you are using OpenCV somewhere in your code and the :class:`~chainer.iterators.MultiprocessIterator` in used in the
```
Comment (kmaehashi) [inline docs/source/tips.rst]: s/in used/is used/

File: docs/source/tips.rst [hunk 3]
```diff
@@ -105,3 +105,17 @@ Now your model is accelerated by iDeep!
...
+training code, the training loop may get stucked at some point. In such situation, there are several workarounds to
+prevent the process got stucked.
+
+1. Set the environment variable as follows: ``OMP_NUM_THREADS=1``
+2. Add ``cv2.setNumThreads(0)`` right after ``import cv2`` in your training script.
+3. Use :class:`~chainer.iterators.MultithreadIterator` instead of :class:`~chainer.iterators.MultiprocessIterator`.
+
+This problem is originally reported here: `A training loop got stuck in a certain condition with multi-processing updater and opencv <https://github.com/chainer/chainer/issues/2903>`_
+for Chainer and the discussion on related problems is still going here `OpenCV + Python multiprocessing breaks on OSX <https://github.com/opencv/opencv/issues/5150>`_.
```
Comment (kmaehashi) [inline docs/source/tips.rst]: s/here/here:/

**Comment by niboshi** [comment]
If #4607 is to be implemented, I think it's better to document it in MultiprocessIterator's doc page.
How do you think?

**Comment by mitmul** [comment]
@kmaehashi Thanks for the review. I fixed all the points you mentioned.
@niboshi Maybe right, but I think it's OK to put this information to both. At this moment, I think merging this for the next release looks fine.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**